public/app/features/plugins/admin/helpers.ts

```diff
@@ -223,13 +223,10 @@ export const sortPlugins = (plugins: CatalogPlugin[], sortBy: Sorters) => {
 };
 
 function groupErrorsByPluginId(errors: PluginError[] = []): Record<string, PluginError | undefined> {
-  return errors.reduce(
-    (byId, error) => {
-      byId[error.pluginId] = error;
-      return byId;
-    },
-    {} as Record<string, PluginError | undefined>
-  );
+  return errors.reduce<Record<string, PluginError | undefined>>((byId, error) => {
+    byId[error.pluginId] = error;
+    return byId;
+  }, {});
```

Comment:
"reduce considered harmful" 😭

Reply:
Can you elaborate on that? I'm asking for learning purposes :)

Reply:
Ash and I have already discussed this, and our opinion differs, so this was mostly unconstructive snark 

But - I think reduce is hard to understand. It trades readability for terseness, prioritising putting everything on one line at the expense of writing code that people can read easily.  

This is also an especially problematic use of reduce as the callback function mutates its argument. It uses the functional array functions in a non functional way. It doesn’t even _reduce_ a value - it converts it to a completely different format. 

I think this would be easier to understand if was just a for loop that mutates an object. Or use lodash’s groupBy. 

(None of this is real feedback for this PR. This function shouldn’t be refactored further in this PR)